R/set_and_assert_intervals.R

```diff
@@ -0,0 +1,52 @@
+#'  Set Intervals
+#'
+#'  Takes in two objects, the PKNCAdata object and the proposed intervals.
+#'  It will then check that the intervals are valid, given the data object.
+#'  If the intervals are valid, it will set them in the object.
+#'  It will return the data object with the intervals set.
+#'  
+#' @param data PKNCAdata object
+#' @param intervals Proposed intervals
+#' @returns The data object with the intervals set.
+#'   
+#' @export
+set_intervals <- function(data, intervals) {
+  valid_intervals <- assert_intervals(intervals, data)
+  
+  data$intervals <- valid_intervals
+  
+  return(data)
```

Comment:
Please use the implicit return code style (remove the `return()` and just use `data`) rather than using `return(data)`.